okr_app/views.py

```diff
@@ -42,3 +45,37 @@ def update(self, request, *args, **kwargs):
             if request.data.get('issuer', None):
                 del request.data['issuer']
         return super().update(request, *args, **kwargs)
+
+class OKRFileViewset(viewsets.GenericViewSet,
+                 mixins.CreateModelMixin,
+                 mixins.DestroyModelMixin,
+                 mixins.RetrieveModelMixin,
+                 mixins.ListModelMixin):
+    """
+    """
+    permission_classes = [IsAuthenticated]
+    serializer_class = OKRFileSerializer
+    queryset = OKRFile.objects.all()
+
+    def get_queryset(self):
+        """
+        """
+        is_admin = IsAdminUser.has_permission(None, self.request, self) or IsApplicationAdminUser.has_permission(None, self.request, None)
+        # Get a list of okr files belong to okr instance
+        if self.request.GET.get("okr", None):
+            okr = get_object_or_404(OKR, pk=self.request.GET.get("okr", None))
+            if (is_admin or IsApplicationAdminUser.has_object_permission(None, self.request, okr)):
+                return self.queryset.filter(okr=okr)    
+            raise NotFound(detail="Error 404, page not found", code=404)
+
+        #If request user is either superadmin or okr_admin, give them all
+        if is_admin:
+            return self.queryset
+        #Else return user's owned file only
+        return self.queryset.filter(okr__issuer=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+        # Just in case we need a list of files uploaded by user
+        request.data['issuer'] = request.user.username
+        return super().create(request, *args, **kwargs)
```

Comment:
```
user = self.request.user
okr = OKR.objects.get(pk=self.kwargs['okr'])
if IsApplicationAdminUser(user) or IsOKROwner(user, okr):
      return self.queryset.filter(okr_id=okr.id)

return self.queryset.none()
```

Reply:
listen to my instruction. Use this logic for queryset, then tell me what the difference between my code and your code, and find out by yourself where you were wrong.